contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -688,6 +690,31 @@ class Resource extends mix(APIResource, IndexedDBResource) {
   }
 }
 
+/**
+ * Tree resources mixin
+ */
+class TreeResource extends Resource {
```

Comment:
Could avoid the subclassing here and just pass `treeLock` method as an option to the Resource constructor for ContentNode - and add in an existence check for `this._locks` - but class extension works too!